aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -105,14 +104,11 @@ public static AppSyncStep builder() {
     /**
      * Start subscribing to model mutations.
      */
-    synchronized void startSubscriptions() {
+    synchronized void startSubscriptions(Action onPipelineBroken) {
         int subscriptionCount = modelProvider.modelNames().size() * SubscriptionType.values().length;
         // Create a latch with the number of subscriptions are requesting. Each of these will be
         // counted down when each subscription's onStarted event is called.
         CountDownLatch latch = new CountDownLatch(subscriptionCount);
-        // Need to create a new buffer so we can properly handle retries and stop/start scenarios.
-        // Re-using the same buffer has some unexpected results due to the replay aspect of the subject.
-        buffer = ReplaySubject.create();
```

Comment:
Before making any of the changes in this PR, I was _not_ able to reproduce this crash if only one subscription is active (just ON_CREATE, for 1 model).  The basic issue causing the crash is due to the following sequence of events:

 1. Going offline closes the socket, which calls `AmplifyWebSocketListener::onFailure`
 2. `onSubscriptionFailure` is called for the first subscription
 3. The error is passed to the buffer (`buffer.onError()`)
 4. `onPipelineBroken::call` (the Action that the Orchestrator calls on error)
 5. `Orchestrator::stopApiBlocking` is called
 6. `SubscriptionProcessor::stopAllSubscriptionActivity` is called, which disposes of the TWO disposables in SubscriptionProcessor: the `buffer`, and the `subscriptionObservables`.
 7.  `onSubscriptionFailure` is called for the second subscription
 8. At this point, the `buffer` is already disposed, and the app crashes with: `io.reactivex.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with`

This PR solves the problem by removing the buffer entirely, and just streaming `SubscriptionEvent` objects from the subscriptionObservables directly to the `Merger`.  The `SubscriptionProcessor` is now only managing  one `Disposable` (the subscriptionObservable stream) instead of two.   The same crash doesn't occur with the `subscriptionObservable` stream, because there is an explicit check to see if the stream is disposed before emitting errors on it.  I don't think a similar check was possible with the buffer though, since it was a `ReplaySubject`, which I don't believe has a way to check if it has been disposed or not.  

The one side effect of this change is that `SubscriptionEvent` are sent to the Merger as they arrive, instead of waiting for all subscriptions to be started.  I don't believe there is any problem with this, but just calling it out.

Reply:
This comment is out of date, right?

Reply:
Yes, it is out of date.  I put the buffer back, because I believe it is needed to prevent a data inconsistency scenario.      Instead, this PR solves the crash by only using the buffer for SubscriptionEvents, not errors.  Errors are kept out of any RX stream since it may be disposed, and instead sent to a error callback provided by the Orchestrator.